javagen/src/main/java/com/azure/autorest/model/clientmodel/ClientMethod.java

```diff
@@ -303,8 +313,27 @@ public void addImportsTo(Set<String> imports, boolean includeImplementationImpor
         }
 
         if (type == ClientMethodType.LongRunningBeginAsync) {
-            if (((GenericType) this.getReturnValue().getType().getClientType()).getTypeArguments()[0] instanceof GenericType) {
-                imports.add("com.fasterxml.jackson.core.type.TypeReference");
+            if (settings.isFluent()) {
+                if (((GenericType) this.getReturnValue().getType().getClientType()).getTypeArguments()[0] instanceof GenericType) {
+                    imports.add("com.fasterxml.jackson.core.type.TypeReference");
+                }
+            } else {
+                imports.add("com.azure.core.util.serializer.TypeReference");
```

Comment:
Just curious. Is there major difference between between `com.fasterxml.jackson.core.type.TypeReference` and `com.azure.core.util.serializer.TypeReference`? Maybe mgmt can switch to the core one as well?

Reply:
I think we probably prefer using classes from our own packages? I believe the `TypeReference` might be used before because that was the only one available.

Reply:
Yes, same thought. Probably when we start on mgmt, `TypeReference` in core did not exist yet. But if it works, we should switch to it.